packages/errors/src/react/error-page.tsx

```diff
@@ -1,116 +1,200 @@
+
+    const size = sizeOverride ? props.size : observableSize
+    const ctx = { size, themeName }
+
+    return (
+      <Context.Provider value={ctx}>
+        <div
+          {...styles.errors(ctx, hasRenderedOnce)}
+          {...rest}
+          ref={ref as RefFor<'div'>}
+        >
+          {illustration}
+          {heading}
+          {caption}
+          {errorCode}
+          {actions}
+        </div>
+      </Context.Provider>
     )
   }
+) as ErrorPageComponent
+
+const Actions: React.FC<HTMLPropsFor<'div'>> = props => (
+  <Context.Consumer>
+    {ctx => <div {...styles.actions(ctx)} {...props} />}
+  </Context.Consumer>
+)
+
+const Caption: React.FC<HTMLPropsFor<'p'>> = props => (
+  <Context.Consumer>
+    {ctx => <p {...styles.caption(ctx)} {...props} />}
+  </Context.Consumer>
+)
+
+const ErrorCode: React.FC<HTMLPropsFor<'p'>> = props => (
+  <Context.Consumer>
+    {ctx => <p {...styles.errorCode(ctx)} {...props} />}
+  </Context.Consumer>
+)
+
+interface HeadingProps extends HTMLPropsFor<'h1'> {
+  as?: keyof Pick<JSX.IntrinsicElements, 'h1' | 'h2' | 'h3' | 'h4' | 'h5'>
 }
 
-const SearchForm: React.FC<Pick<ErrorPageProps, 'action'>> = props => {
-  const themeName = useTheme()
+const Heading: React.FC<HeadingProps> = props => {
+  const { as: Tag = 'h1', ...rest } = props
+
   return (
-    <form action={props.action} method="get" {...styles.search(themeName)}>
-      <div {...styles.searchIcon(themeName)}>{icons.search()}</div>
-      <input
-        {...styles.searchInput(themeName)}
-        type="text"
-        name="q"
-        placeholder="Search"
-      />
-    </form>
+    <Context.Consumer>
+      {ctx => <Tag {...styles.heading(ctx)} {...rest} />}
+    </Context.Consumer>
   )
 }
 
-export interface ErrorPageProps extends HTMLPropsFor<'div'> {
-  action?: string
-  code?: string
-  href?: string
-  iconId?: keyof typeof icons
-  text?: string
-}
+interface IllustrationProps extends HTMLPropsFor<'svg'> {}
+
+const Illustration: React.FC<IllustrationProps> = props => {
+  const { children: custom, ...rest } = props
 
-const ErrorPage: React.FC<ErrorPageProps> = props => {
-  const { action, code, href, iconId, text, ...rest } = props
-  const themeName = useTheme()
   return (
-    <div {...styles.page(themeName)} {...rest}>
-      {iconId && icons[iconId] && (
-        <div {...styles.icon(themeName)}>{icons[iconId]()}</div>
-      )}
-      <div {...styles.text()}>
-        <Heading size={Heading.sizes.medium}>
-          <h1>{text}</h1>
-        </Heading>
-      </div>
-      <div {...styles.code(themeName)}>
-        <h2>
-          <Label size={Label.sizes.xSmall} color={Label.colors.secondary}>
-            Error code: {code}
-          </Label>
-        </h2>
-      </div>
-      {href && <Button href={href}>Contact support</Button>}
-      {action && <SearchForm action={action} />}
-    </div>
+    <Context.Consumer>
+      {ctx => {
+        // @ts-ignore: necessary conditional
```

Comment:
What's the necessary conditional?  (I don't see one on the next line.). And why does it need ignored?